### ISSUE TITLE
Avoid the need to set a different adapter when establishing a connection

### ISSUE THREAD

**Issue by leoasis** [leoasis/activerecord-futures#9]
It would be nice to not have to define a custom adapter key when establishing a connection in activerecord. Since the normal case when you install this gem is that you WANT to have futures enabled automatically. It would not make sense to want to use the adapter without futures.

So, perhaps we could hook into the connection specification class and provide a different spec when it is an adapter that we support.

It would also be nice to directly support defining the real adapter names as it is working now.

Basically the idea is that with `adaper: mysql2` and this gem installed, you automatically have the `FutureEnabledMysql2Adapter` as the ActiveRecord connection. Same for postgres. Also, this should be equivalent to specifying `adapter: future_enabled_mysql`, which should still be working.
